tests/test_x509.py

```diff
@@ -173,6 +174,24 @@ def test_revoked_cert_retrieval(self, backend):
         # Check that len() works for CRLs.
         assert len(crl) == 12
 
+    def test_revoked_cert_retrieval_retain_only_revoked(self, backend):
+        """
+        This test attempts to trigger the crash condition described in
+        https://github.com/pyca/cryptography/issues/2557
+        """
+        crl = _load_cert(
+            os.path.join("x509", "custom", "crl_all_reasons.pem"),
+            x509.load_pem_x509_crl,
+            backend
+        )
+        revoked = crl[11]
+        crl = "overwritten"
```

Comment:
IMO it's cleaner to just write `revoked = _load_cert(...)[11]` that way there's no confusing unused writes to locals.